src/main/java/edu/kit/compiler/semantic/NamespaceMapper.java

```diff
@@ -0,0 +1,42 @@
+package edu.kit.compiler.semantic;
+
+import edu.kit.compiler.data.ast_nodes.ClassNode;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NamespaceMapper {
+
+    private final Map<ClassNode, ClassNamespace> symbolTableMap = new HashMap<>();
+
+    @AllArgsConstructor
+    @Data
+    public static class ClassNamespace {
+        private SymbolTable methodSymbols;
+        private SymbolTable fieldSymbols;
```

Comment:
Do we need a full fledged `SymbolTable` here? This is a simple mapping between an identifier and a field or method respectively. Would an ordinary `Map` not do the job just as well?

Reply:
Also, we could map to `MethodNode`/`ClassNodeField` directly to gain access to e.g. method argument types without having to cast the returned `Definition`.

Reply:
@g4rgoil 
Yes I think we can shorten the definition here. The mapper is only used for looking up if a method or field is defined anyway and give the correct. So the mapper holds one symbol table per class and defined methods and fields